model/map/src/main/java/org/keycloak/models/map/clientscope/MapClientScopeAdapter.java

```diff
@@ -142,26 +163,40 @@ public void removeProtocolMapper(ProtocolMapperModel mapping) {
     public void updateProtocolMapper(ProtocolMapperModel mapping) {
         final String id = mapping == null ? null : mapping.getId();
         if (id != null) {
-            entity.updateProtocolMapper(id, mapping);
+            if (entity.getProtocolMapper(id).isPresent()) {
+                entity.removeProtocolMapper(id);
+                addProtocolMapper(mapping);
+            }
         }
     }
 
     @Override
     public ProtocolMapperModel getProtocolMapperById(String id) {
-        return entity.getProtocolMapperById(id);
+        if (entity.getProtocolMapper(id).isPresent()) {
+            return pmUtils.toModel(entity.getProtocolMapper(id).get());
+        }
+        return null;
```

Comment:
`entity.getProtocolMapper(id)` is called twice here. Better leverage available `Optional`:
```suggestion
        return entity.getProtocolMapper(id).map(pmUtils::toModel).orElse(null);
```